python/ray/serve/tests/unit/test_deployment_scheduler.py

```diff
@@ -788,6 +807,68 @@ def test_heterogeneous_resources(self):
             assert scheduling_strategy.node_id == "node1"
             assert call.kwargs == {"placement_group": None}
 
+    @pytest.mark.parametrize("use_pg", [True, False])
```

Comment:
I think currently max_replicas_per_node doesn't work with pg (The semantic is only clear when it's STRICT_PACK).

Let me actually create a PR to throw an error when both max_replicas_per_node and pg are specified.

Reply:
Yeah `use_pg` actually means `use_strict_pack_pg`, let me rename it.

Reply:
Synced offline, removing this test and making an assumption that max_replicas_per_node and placement_group_bundles will not both be specified and do not work together.

Reply:
will these not work together even if its strict pack?

Reply:
Also if this is the case let's make sure it's documented & raises a good error

Reply:
Yeah I think @jjyao is working on a PR to add validation + throw good error when user tries to specify both.

Reply:
We haven't really documented pg btw and the main use is in rayllm where it's always strict pack

Reply:
In theory, max_replicas_per_node can work with strict_pack pg but given rayllm doesn't use max_replicas_per_node, we can add the support later. For now, I'll just create a PR to make these two options mutually exclusive.